app/services/time_limit_calculator.rb

```diff
@@ -7,16 +7,35 @@ def initialize(rule:, effective_date:)
   end
 
   def call
-    days = calculate_days
+    days = calculate_day
+
+    if beyond_eoc(days)
+      calculate_eoc_rbd
+    else
+      {
+        days: days,
+        time_in_future: calculate_time_in_future(days),
+        time_in_past: calculate_time_in_past(days),
+      }
+    end
+  end
+
+private
+
+  def beyond_eoc(days)
+    calculate_time_in_future(days) >= CycleTimetable.find_closes
+  end
+
+  def calculate_eoc_rbd
+    days = effective_date.to_date.business_days_until((CycleTimetable.find_closes - 2.days).to_date)
```

Comment:
What is the `2.days` for?

Reply:
Originally had `- 1.day`,  as the EoC RBD date is intended to be the day before the deadline date. 

`1.day` was causing the specs to fail whereas they pass with `2.days`, although I worry this is perhaps a bit 2021 cycle specific....

Reply:
think this is resolved now!